### PR TITLE
refactor(typescript): packages/rspack/src/logging/runtime

### PR DESCRIPTION
**Pull request by wxiaoyun**
## Summary

- Improve typescript coverage
- Addresses https://github.com/web-infra-dev/rspack/issues/7225

## Checklist

- [X] Tests updated (or not required).
- [X] Documentation updated (or not required).

### CODE REVIEW

File: packages/rspack/src/logging/type.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	profile?: (...args: any[]) => void;
+	profileEnd?: (...args: any[]) => void;
+	logTime?: (...args: any[]) => void;
+};
+
+export type LoggerOptions = {
+	level: "none" | "error" | "warn" | "info" | "log" | "verbose" | boolean;
+	debug: FilterTypes | boolean;
+	console: LoggerConsole;
+};
```
Comment (wxiaoyun) [inline packages/rspack/src/logging/type.ts:36]: Cannot export types from createConsoleLogger.ts since its using module exports.

Comment (SoonIter) [reply packages/rspack/src/logging/type.ts:36]: could you migrate `createConsoleLogger.js` to esm ?

Comment (wxiaoyun) [reply packages/rspack/src/logging/type.ts:36]: Sure

File: packages/rspack/src/logging/createConsoleLogger.ts [hunk 2]
```diff
@@ -81,25 +47,14 @@ const LogLevel = {
 	verbose: 1
 };
 
-/**
- * @param {LoggerOptions} options options object
- * @returns {function(string, LogTypeEnum, any[]): void} logging function
- */
-export = ({ level = "info", debug = false, console }: any) => {
+export = ({ level = "info", debug = false, console }: LoggerOptions) => {
```
Comment (SoonIter) [inline packages/rspack/src/logging/createConsoleLogger.ts:50]: I think it is okay for you to migrate it to
`export { createConsoleLogger }`
you can have a try

**Review by SoonIter — APPROVED** [review]
LGTM, ❤️

**Pull request merged**